### DISCUSSION TITLE
Clarification on API Key Requirement for INSPIREMetadataValidator

### DISCUSSION THREAD

**Discussion by Runge123** [eblondel/geometa#210] [Q&A]
I am using the INSPIREMetadataValidator class to validate metadata against INSPIRE standards using the geometa package. According to the documentation, it seems necessary to provide an apiKey when initializing an object of this class. However, I cannot find clear instructions on how to obtain this key from the INSPIRE Validator.

Steps to Reproduce:

> Instantiate an object of INSPIREMetadataValidator without providing an API key.

> Call the getValidationReport() method with an ISOMetadata object.

> Encounter the following error:
Error in initialize(...): argument "apiKey" is missing, with no default

Questions:

How can I obtain an apiKey for the INSPIRE Validator?
Is there an alternative way to perform validation without an API key in development or testing environments?

Thank you for your assistance :D

**Comment by eblondel** [comment]
Hello @Runge123 There is no alternative way to perform validation without an API key, unless you want to manually check your metadata within the INSPIRE web metadata validator.

In geometa, this is entirely delegated to the INSPIRE web-services provided by the INSPIRE team. To get an API key, you need to contact the INSPIRE team at [jrc-inspire-support@ec.europa.eu](mailto:jrc-inspire-support@ec.europa.eu); For help you can use this helpdesk: https://github.com/INSPIRE-MIF/helpdesk-validator/issues
